prow/github/client.go

```diff
@@ -2475,6 +2482,34 @@ func (c *client) GetBranches(org, repo string, onlyProtected bool) ([]Branch, er
 	return branches, nil
 }
 
+// GetDefaultBranch returns default branch of the repo.
+// Uses a memory cache for performance, storing repos with key `org_repo_defaultBranch`.
+//
+// See https://docs.github.com/en/rest/reference/repos#list-organization-repositories
+func (c *client) GetDefaultBranch(org, repo string) (string, error) {
```

Comment:
Nit, could we make sure to clarify that this is cached?
```suggestion
func (c *client) GetDefaultBranchCached(org, repo string) (string, error) {
```